ThunderRequest/UIAlertController+ErrorRecovery.swift

```
@@ -0,0 +1,109 @@
+            if !message.isEmpty {
+                message.append("\n")
+            }
+            message.append(recoverySuggestion)
+        }
+        
+        return message.isEmpty ? nil : message
+    }
+}
+
+extension UIAlertController {
+    
+    /// Presents an error as a recoverable error from the given view controller
+    ///
+    /// - Parameters:
+    ///   - error: The error to present
+    ///   - viewController: The view controller to present it in
+    public static func present(error: Error, in viewController: UIViewController) {
+        
+        let alertController = UIAlertController(error: error)
+        OperationQueue.main.addOperation { [weak viewController] in
+            viewController?.present(alertController, animated: true, completion: nil)
+        }
+    }
+    
+    /// Initialises a UIAlertController from a given error
+    ///
+    /// This will add recovery options if the error conforms to `CustomisableRecoverableError` or `RecoverableError`
+    ///
+    /// - Parameter error: The error to show
+    public convenience init(error: Error) {
+        
+        let errorOverride = ErrorOverrides.overrideFor(domain: (error as NSError).domain, code: (error as NSError).code)
+        
+        self.init(
+            title: errorOverride.description ?? (error as? CustomisableRecoverableError)?.description ?? error.localizedDescription,
+            message:  ErrorOverrides.recoveryMessageFor(error: error),
+            preferredStyle: .alert
+        )
+        
+        switch error {
+        case var anyRecoverableError as CustomisableRecoverableError:
+            
+            if anyRecoverableError.options.isEmpty {
+                anyRecoverableError.add(option: ErrorRecoveryOption(title: "Dismiss", style: .cancel))
+            }
+            
+            anyRecoverableError.options.enumerated().forEach { (index, option) in
```

Following the same logic as the case block below, if the options are empty is it worth doing this? The above could be a guard instead & just return out before doing this :)

---

I think we'd then be hiding errors from the user that the consumer of this API clearly wanted them to see. I wouldn't expect to call `UIAlertController(error: _someError)` and have it return a broken `UIAlertController` :)